mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -56,13 +56,13 @@ impl PeerManager {
     }
 
     // returns true if `peer` is discovered for first time, else false
-    pub fn add_peer(&self, peer: PeerNetworkId) -> bool {
+    pub fn add_peer(&self, peer: Arc<PeerNetworkId>) -> bool {
```

Comment:
probably don't need Arc here and would argue for simplicity implementing Copy on NetworkId and NetworkContext is probably better at this point in time.

Reply:
yeah, it's unfortunate that NetworkId has a String so implementing Copy is kind of a pain :/